benchmark/Main.hs

```diff
@@ -2,30 +2,38 @@
 
 module Main where
 
-import Criterion.Main (defaultMain)
+import Criterion.Main (defaultMain, bgroup)
 
-import CacheMap (benchCacheMap)
-import CMap (benchMap, prepareBenchMap)
+import Common
+import qualified CMap
+import qualified CacheMap
 #if ( __GLASGOW_HASKELL__ >= 802 )
-import DMap (benchDMap, prepareBenchDMap)
+import qualified DMap
 #endif
-import OptimalVector (benchVectorOpt, prepareBenchVectorOpt)
---import Vector (benchVector, prepareBenchVector)
+import qualified OptimalVector as OptVec
 
 main :: IO ()
 main = do
-  prepareBenchMap
-  --prepareBenchVector
-  prepareBenchVectorOpt
+  let spec = [("CMap", CMap.spec)
+             ,("CacheMap", CacheMap.spec)
 #if ( __GLASGOW_HASKELL__ >= 802 )
-  prepareBenchDMap
+             , ("DMap", DMap.spec)
 #endif
+             , ("OptVec", OptVec.spec)
+             ]
+      mkGroup f = 
+        [ b label
+        | (label, v) <- spec 
+        , Just b <- pure $ f v
```

Comment:
Could you please give more descriptive names to the variables or even add types to the functions? It's very hard for me to keep in my head that much stuff, sorry.

Reply:
ok